projects/com.oracle.truffle.llvm.parser.bc/src/com/oracle/truffle/llvm/parser/bc/LLVMBitcodeVisitor.java

```diff
@@ -348,10 +359,74 @@ private LLVMExpressionNode getGlobalVariable(GlobalValueSymbol global) {
             final LLVMExpressionNode loadedFunction = factoryFacade.createLoad(this, functionType, functionLoadTarget);
             final LLVMExpressionNode[] argNodes = new LLVMExpressionNode[]{factoryFacade.createFrameRead(this, LLVMBaseType.ADDRESS, stackSlot)};
             final LLVMNode functionCall = factoryFacade.createFunctionCall(this, loadedFunction, argNodes, LLVMBaseType.VOID);
-            structors[i] = functionCall;
+
+            final StructureConstant structureConstant = (StructureConstant) arrayConstant.getElement(i);
+            final Integer priorityConstant = LLVMSymbolResolver.evaluateIntegerConstant(structureConstant.getElement(0));
+            assert priorityConstant != null;
+            sortedStructors.add(priorityConstant, functionCall);
+        }
+
+        return sortedStructors.getAll();
+    }
+
+    private static final class SortedLLVMNodeList {
```

Comment:
@jkreindl, please do not introduce a new collection implementation. As Manuel stated, this is error-prone.
If I understand your code correctly, you want to assign a priority to an `LLVMNode` and sort by this priority.

One trivial solution could be that you have 
```
class PriorityElement {
  final LLVMNode node;
  final int priority;
}

```
and have this in a sorted Collection. Use a comparator to sort by priorities.